include/odp/api/spec/dma_types.h

```diff
@@ -470,26 +470,64 @@ typedef struct odp_dma_seg_t {
 	/** Segment hints
 	 *
 	 *  Depending on the implementation, setting these hints may improve performance.
-	 *  Initialize all unused bits to zero.
 	 */
-	union {
-		/** Segment hints bit field */
+	struct {
+		/** Cache stashing hints
+		 *
+		 *  Applicable only to destination segments. Ignored for source segments. Hints
+		 *  implementation to bring specififed cache lines into cache as part of successful
+		 *  transfer completion.
+		 */
 		struct {
-			/** Allow full cache line access
+			/** L3 cache stashing */
+			struct {
+				/** Byte offset into the segment data to start caching from
+				 *
+				 *  Depending on the implementation, this might be rounded down
+				 *  to a more suitable boundary. In case of #ODP_DMA_FORMAT_PACKET,
+				 *  offset is in addition to the packet data starting offset.
+				 */
+				uint32_t offset;
+
+				/** Length in bytes to cache
+				 *
+				 *  Depending on the implementation, this might be rounded up
+				 *  to a more suitable boundary.
+				 */
+				uint32_t len;
+			} l3;
+		} cache_stash;
+
+		union {
+			/** Segment hints bit field
+			 *
+			 *  Initialize all unused bits to zero.
+			 */
+			struct {
```

Comment:
A matter of taste, but perhaps it would be a bit clearer when reading from top to bottom, if the segment hints bit field union would be the first member of `hints` struct.